src/net.rs

```diff
@@ -0,0 +1,52 @@
+//! Networking-related helper functions.
+
```

Comment:
So, this could/should actually live in `vector-core`.

Ironically, `vector-core` already has a nearly identical file in there doing almost the exact same thing, but when I went to convert it all over, it felt like a lot of toil/change on top of an already large PR, and I felt like it would be an easy lay-up PR to follow up with.